src/pip/_internal/req/req_set.py

```diff
@@ -122,7 +119,8 @@ def add_requirement(
             return [install_req], None
 
         try:
-            existing_req = self.get_requirement(install_req.name)
+            existing_req = self.get_requirement(
+                install_req.name)  # type: Optional[InstallRequirement]
         except KeyError:
             existing_req = None
```

Comment:
nit: I'd prefer a slightly different code flow, over this awkward line break.

```py
        existing_req = None   # type: Optional[InstallRequirement]
        try:
            existing_req = self.get_requirement(install_req.name)
        except KeyError:
            pass
```

Reply:
It's perfectly OK to not do this though. This is definitely more of a personal preference thing. :)

Reply:
Okay, then I will keep this as is.